src/DataBuilder.php

```diff
@@ -754,13 +754,18 @@ protected function scrub($arr, $fields, $replacement = '*')
             return null;
         }
 
-        $scrubber = function (&$val, $key) use ($fields, $replacement, $arr) {
-
+        $scrubber = function (&$val, $key) use ($fields, $replacement, &$scrubber) {
             if (in_array($key, $fields)) {
                 $val = str_repeat($replacement, 8);
             }
+
+            if (is_array($val)) {
+                array_walk($val, $scrubber);
+            }
         };
-        array_walk_recursive($arr, $scrubber);
```

Comment:
This is confusing to me. I don't understand the reasoning behind doing a manual recursion versus array_walk_recursive()